projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -122,6 +123,11 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseDirecti
      */
     public parent = null;
 
+    /**
+     * @hidden
+     */
+    protected _rowSelectionMode: GridSelectionMode;
```

Comment:
You wouldn't need this.

Reply:
This is is overridden just so the getter doesn't complain about the return type, though (if the types are kept) this is easily adjusted in the getter as well with a type assert.

Reply:
@damyanpetev I don't want a new enum type though, and so this won't be necessary.